packages/runtime/container-runtime/src/opLifecycle/outbox.ts

```diff
@@ -259,18 +260,49 @@ export class Outbox {
 			true /* disableGroupedBatching */,
 			resubmittingBatchId,
 		);
-		this.flushInternal(
+		this.flushInternalMainBatch(
 			this.mainBatch,
 			false /* disableGroupedBatching */,
 			resubmittingBatchId,
 		);
 	}
 
+	/**
+	 * Flushes the main batch to the ordering service. If the batch is empty, it will create an empty grouped batch
+	 * if BatchId is present (resubmit flow) and the runtime should send.
+	 */
+	private flushInternalMainBatch(
+		mainBatch: BatchManager,
+		disableGroupedBatching: boolean = false,
+		resubmittingBatchId?: BatchId,
+	) {
+		let clientSequenceNumber: number | undefined;
+		if (mainBatch.empty) {
+			if (resubmittingBatchId && this.params.config.immediateMode !== true) {
```

Comment:
Why the immediateMode check?

Reply:
Add an assert before to check if offlineLoadEnabled is true and immediateMode is true -> close the container.